src/ui/robot/Robot.tsx

```diff
@@ -64,6 +64,17 @@ const sortTree = (roots: FileTreeEntry[], sorter: (a: FileTreeEntry, b: FileTree
 export function Robot({ project }: { project: Project }) {
   const [selectedFile, setSelectedFile] = useState(project.generatedFiles.find(f => f.name === ROBOT_CLASS_PATH))
 
+  // Reload the current file when the project changes
+  useEffect(() => {
+    const currentFile = selectedFile
+    setSelectedFile(project.generatedFiles.find(f => f.name === "README.md"))
+
+    if (project.generatedFiles.find(f => f.name === currentFile.name)) {
+      // Show the original file if it still exists
+      setSelectedFile({ ...currentFile })
+    }
+  }, [project])
```

Comment:
For some reason, this only seems to fire once, and requires a view reload (such as switching tabs) afterward